playwright/_impl/_element_handle.py

```diff
@@ -343,9 +342,25 @@ async def wait_for_selector(
         state: Literal["attached", "detached", "hidden", "visible"] = None,
         timeout: float = None,
         strict: bool = None,
-    ) -> Optional["ElementHandle"]:
-        return from_nullable_channel(
-            await self._channel.send("waitForSelector", locals_to_params(locals()))
+    ) -> "ElementHandle":
+        return cast(
+            ElementHandle,
+            from_nullable_channel(
```

Comment:
```suggestion
            from_channel(
```

Also `cast` is redundant here